src/hooks/useBlockPosition.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+
+export const useBlockPosition = (ref: any) => {
```

Comment:
What is it?

Reply:
I created a hook to encapsulate the logic for calculating the window size and tooltip, I will remove it